addons/website_blog/models/website_blog.py

```diff
@@ -36,111 +33,80 @@ def all_tags(self, cr, uid, ids, min_limit=1, context=None):
-        'content': fields.html('Content', translate=True, sanitize=False),
-        'website_message_ids': fields.one2many(
-            'mail.message', 'res_id',
-            domain=lambda self: [
-                '&', '&', ('model', '=', self._name), ('type', '=', 'comment'), ('path', '=', False)
-            ],
-            string='Website Messages',
-            help="Website communication history",
-        ),
-        # creation / update stuff
-        'create_date': fields.datetime(
-            'Created on',
-            select=True, readonly=True,
-        ),
-        'create_uid': fields.many2one(
-            'res.users', 'Author',
-            select=True, readonly=True,
-        ),
-        'write_date': fields.datetime(
-            'Last Modified on',
-            select=True, readonly=True,
-        ),
-        'write_uid': fields.many2one(
-            'res.users', 'Last Contributor',
-            select=True, readonly=True,
-        ),
-        'author_avatar': fields.related(
-            'author_id', 'image_small',
-            string="Avatar", type="binary"),
-        'visits': fields.integer('No of Views'),
-        'ranking': fields.function(_compute_ranking, string='Ranking', type='float'),
-    }
-
-    _defaults = {
-        'name': '',
-        'content': _default_content,
-        'cover_properties': '{"background-image": "none", "background-color": "oe_none", "opacity": "0.6", "resize_class": ""}',
-        'author_id': lambda self, cr, uid, ctx=None: self.pool['res.users'].browse(cr, uid, uid, context=ctx).partner_id.id,
-    }
-
-    def html_tag_nodes(self, html, attribute=None, tags=None, context=None):
+    @api.multi
+    @api.depends('visits')
+    def _compute_ranking(self):
+        for blog_post in self:
+            if blog_post.visits:
+                age = datetime.now() - datetime.strptime(blog_post.create_date, tools.DEFAULT_SERVER_DATETIME_FORMAT)
+                blog_post.ranking = blog_post.visits * (0.5+random.random()) / max(3, age.days)
+
+    name = fields.Char(string='Title', required=True, translate=True, default='')
+    subtitle = fields.Char(string='Sub Title', translate=True, default=_('Subtitle'))
+    author_id = fields.Many2one('res.partner', string='Author', default=lambda self:self.env.user.partner_id)
+    cover_properties = fields.Text(string='Cover Properties', default='{"background-image": "none", "background-color": "oe_none", "opacity": "0.6", "resize_class": ""}')
+    blog_id = fields.Many2one(
+        'blog.blog', string='Blog',
+        required=True, ondelete='cascade'
+    )
+    tag_ids = fields.Many2many(
+        'blog.tag', string='Tags'
+    )
+    content = fields.Html(string='Content', translate=True, sanitize=False, default=_default_content)
+    website_message_ids = fields.One2many(
+        'mail.message', 'res_id',
+        domain=lambda self: [
+            '&', '&', ('model', '=', self._name), ('type', '=', 'comment'), ('path', '=', False)
+        ],
```

Comment:
hello @atp-odoo , i doesn't think that we should remove the domain of path because we have used "website_message_ids" at more than one place and at every place  need to do filtered(lambda post: post.path == False)  so instead of getting data and then filtering that data just passing domain is better thing right?? Thanks